src/Layout/QueryParamContext.tsx

```diff
@@ -219,8 +220,21 @@ export default function QueryParamProvider({ children }: any) {
     }
   }, [paymentError]);
   React.useEffect(() => {
-    if (router.query.locale) {
+    if (allLocales.find((locale) => locale.key === router.query.locale)) {
       setlanguage(router.query.locale);
+    } else {
+      //conditional operator to check if navigator.languages property is supported by browser.
+      const userLocale =
+        navigator.languages === undefined
+          ? [navigator.language]
+          : navigator.languages;
+      const newLocale = userLocale[0].trim().split(/-|_/)[0];
+
+      if (allLocales.find((locale) => locale.key === newLocale)) {
```

Comment:
Though `.find()` will result into an object which will coerce into a boolean, that will get the job done.
But I think we should use `.some()` here.